schema/config.d.ts

```diff
@@ -0,0 +1,61 @@
+import { Feature, SiteException } from './feature';
+import { AutoconsentFeature } from './features/autoconsent';
+import { CookieFeature } from './features/cookie';
+import { TrackerAllowlistFeature } from './features/tracker-allowlist';
+import { WebCompatFeature } from './features/webcompat';
+import { DuckPlayerFeature } from './features/duckplayer';
+
+export { WebCompatSettings } from './features/webcompat';
+export { DuckPlayerSettings } from './features/duckplayer';
+
+export type ExportedSchemas =
+    | 'GenericV4Config'
+    | 'AndroidV4Config'
+    | 'LegacyConfig'
+    | 'LegacyAndroidConfig'
+    | 'WebCompatSettings'
+    | 'DuckPlayerSettings';
+
+/**
+ * Defines the structure of the built V4 config output as downloaded by clients.
+ */
+export type ConfigV4<VersionType> = {
+    readme: string;
+    version: number;
+    features: Record<string, Feature<any, VersionType>> & {
+        // These features have typed settings
+        autoconsent: AutoconsentFeature<VersionType>;
+        cookie: CookieFeature<VersionType>;
+        duckPlayer: DuckPlayerFeature<VersionType>;
+        trackerAllowlist: TrackerAllowlistFeature<VersionType>;
+        webCompat: WebCompatFeature<VersionType>;
```

Comment:
Just so I understand: Does the webcompat-test.js add anything else type wise?  (Same for duckplayer)

Reply:
No - we're just applying a subset of the schema on a subset of the data. I just kept the existing tests there, but the full schema test will be testing the same thing now, so these are redundant.